src/main/controller/PromotionController.ts

```diff
@@ -109,24 +103,29 @@ export class PromotionController {
           { abortEarly: false }
         );
 
+        // find existing restaurant
+        const restaurant = await transactionalEntityManager
+          .getCustomRepository(RestaurantRepository)
+          .findOne({ placeId: promotionDTO.placeId });
         const user = await transactionalEntityManager
           .getCustomRepository(UserRepository)
           .findOneOrFail(promotionDTO.userId);
         const promotion = DTOConverter.promotionDTOtoPromotion(
           promotionDTO,
-          user
+          user,
+          // todo PP-82 if restaurant does not exist create a new one with correct lat and lon
+          restaurant ??
+            new Restaurant(
+              promotionDTO.placeId,
+              randomLatitude(),
+              randomLongitude()
+            )
```

Comment:
@ld357 important part when u rebase ur branch, this is the only place I think that should conflict with ur work